apps/server/src/modules/me/api/mapper/me.response.mapper.ts

```diff
@@ -0,0 +1,124 @@
+import { School } from '@src/modules/school';
+import { Role, User } from '@shared/domain/entity';
+import { EntityId } from '@shared/domain/types';
+import {
+	MeAccountResponse,
+	MeResponse,
+	MeRolesReponse,
+	MeSchoolLogoResponse,
+	MeSchoolResponse,
+	MeUserResponse,
+} from '../dto';
+
+export class MeResponseMapper {
+	public static mapToResponse(school: School, user: User, accountId: EntityId): MeResponse {
+		const schoolResponse = MeResponseMapper.mapSchool(school);
+		const userResponse = MeResponseMapper.mapUser(user, school);
+		const rolesResponse = MeResponseMapper.mapUserRoles(user);
+		const permissionsResponse = MeResponseMapper.mapPermissions(user);
+		const accountResponse = MeResponseMapper.mapAccount(accountId);
+
+		const res = new MeResponse({
+			school: schoolResponse,
+			user: userResponse,
+			roles: rolesResponse,
+			permissions: permissionsResponse,
+			account: accountResponse,
+		});
+
+		return res;
+	}
+
+	private static mapSchool(school: School): MeSchoolResponse {
+		const schoolInfoProps = school.getInfo();
+		// i missed the school logoName and url ..is used in frontend but unkown on which place but should expose over me
+		// please check vue header components with ticket BC-6371
+		const logo = new MeSchoolLogoResponse({
+			url: schoolInfoProps.logo_dataUrl || null,
+			name: schoolInfoProps.logo_name || null,
+		});
+
+		const schoolResponse = new MeSchoolResponse({
+			id: schoolInfoProps.id,
+			name: schoolInfoProps.name,
+			logo,
+		});
+
+		return schoolResponse;
+	}
+
+	// comments inside the method are for connect it with vue, remove it with ticket BC-6371
+	private static mapUser(user: User, school: School): MeUserResponse {
+		const userInfo = user.getInfo();
+		const schoolInfoProps = school.getInfo();
+		// v1 - user go out with all infos, for example
+		// - fullName | displayName -> why? -> frontend!
+		// - consent
+		// - birthday
+		// - preferences
+		// - externallyManaged -> store.userIsExternallyManaged ..not in use and solving externallyManaged is total wrong placed in me
+		// - avatarInitials -> First char of firstName and lastName -> frontend!
+		// - avatarBackgoundColor --> need to be checked, but look like frontend we should expose customAvatarBackgroundColor
+		// - age
+
+		const userResponse = new MeUserResponse({
+			id: userInfo.id,
+			firstName: userInfo.firstName,
+			lastName: userInfo.lastName,
+			language: userInfo.language || schoolInfoProps.language || null,
+			customAvatarBackgroundColor: userInfo.customAvatarBackgroundColor || null,
+		});
+
+		/* take from featherJS, must be moved to vue
+		const setAvatarData = (user) => {
```

Comment:
Was ist mit diesem kommentierten Code?

Reply:
"take from featherJS, must be moved to vue."
Aus firstName und lastName wird der erste Buchstabe genommen um den Avatar(Namen) zu genieren Marla Mathe ist MM. Das MM wird wiederum genommen um daraus die Avatar Farbe zu genieren, sollte keine defaultFarbe gesetzt sein. 

Das erfolgt momentan im after hook der v1 me Route. Hat da aber eigentlich gar nichts zu suchen. Das kann direkt im client erfolgen über den firstName / lastName. Damit im folge Ticket die Informationen nicht alle neu gesucht werden müssen habe ich das in dem PR schon notiert.

Reply:
verschoben ins Ticket